packages/delivery-expo/delivery.js

```diff
@@ -0,0 +1,138 @@
+const payload = require('@bugsnag/core/lib/json-payload')
+const { isoDate } = require('@bugsnag/core/lib/es-utils')
+const UndeliveredPayloadQueue = require('./queue')
+const NetworkStatus = require('./network-status')
+const RedeliveryLoop = require('./redelivery')
+
+module.exports = (client, fetch = global.fetch) => {
+  const networkStatus = new NetworkStatus()
+
+  const send = (url, opts, cb) => {
+    fetch(url, opts)
+      .then(response => {
+        if (response.ok) return response.text()
+        const err = new Error(`Bad status code from API: ${response.status}`)
+        err.isRetryable = isRetryable(response.status)
+        return Promise.reject(err)
+      })
+      .then(() => cb(null))
+      .catch(err => cb(err))
+  }
+
+  const logError = e => client._logger.error('Error redelivering payload', e)
+
+  const enqueue = async (payloadKind, failedPayload) => {
+    client._logger.info(`Writing ${payloadKind} payloads to cache`)
+    await queues[payloadKind].enqueue(failedPayload, logError)
+    if (networkStatus.isConnected) queueConsumers[payloadKind].start()
+  }
+
+  const onerror = async (err, failedPayload, payloadKind, cb) => {
+    client._logger.error(`${payloadKind} failed to send…\n${(err && err.stack) ? err.stack : err}`, err)
+    if (failedPayload && err.isRetryable !== false) enqueue(payloadKind, failedPayload)
+    cb(err)
+  }
+
+  const { queues, queueConsumers } = initRedelivery(networkStatus, logError, send)
+
+  return {
+    sendReport: (report, cb = () => {}) => {
+      const url = client.config.endpoints.notify
+
+      let body, opts
+      try {
+        body = payload.report(report, client.config.filters)
+        opts = {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'Bugsnag-Api-Key': report.apiKey || client.config.apiKey,
+            'Bugsnag-Payload-Version': '4',
+            'Bugsnag-Sent-At': isoDate()
+          },
+          body
+        }
+        if (!networkStatus.isConnected || (report._handledState && report._handledState.attemptDelivery === false)) {
```

Comment:
Yeah I pondered over where the best place to attach this information to the report was, and of all the existing places I think it's here. But I agree it's not perfect.

Basically the `react-native-global-error-handler` needs to communicate _"I caught this error which would have caused the app to reload"_. `unhandled: true` is the case for stuff like render errors caught in an react error boundary, which wouldn't cause that, so we can't just base this off of `unhandled` unfortunately – we need the extra info.

Reply:
My gut feeling is that it would make more sense as `report.attemptDelivery`, as whether delivery should be attempted is a property of a report, rather than its handledState.

Is there any risk that including this value in the `report` could result in it being serialised?

Reply:
> Is there any risk that including this value in the report could result in it being serialised?

Nope, we manually describe the properties that get serialised [here](https://github.com/bugsnag/bugsnag-js/blob/2091b7e62e2e073bcbbd5f90d3ff29a2d813672a/packages/core/report.js#L94-L118).

I think I'd be happy to move to your alternative suggestion so I'll make that change.